test/unit/host_discovered_test.rb

```diff
@@ -148,6 +168,18 @@ class HostDiscoveredTest < ActiveSupport::TestCase
     assert Token.where(:host_id => h.id).empty?
   end
 
+  test "discovered host hostname normalization" do
+    assert_equal Host::Discovered.normalize_string_for_hostname("90:B1:1C:54:D5:82"),"90b11c54d582"
+    assert_equal Host::Discovered.normalize_string_for_hostname(".-_Test::Host.name_-."),"testhostname"
+    assert_equal Host::Discovered.normalize_string_for_hostname("testhostname"),"testhostname"
+  end
```

Comment:
Please note that underscores are not allowed in hostnames.
Also, how can the normalize_string_for_hostname return nil or empty string, if we're going to make it throw exception in such cases? I can however make a test that checks it does throw an exception in such cases.